infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -480,4 +481,21 @@ public static void addManagementServer() {
         managementServer = new ManagementServer(serverContext);
         router.addServer(managementServer);
     }
+
+    /** Generate a Node Id if not present.
+     *
+     * @param context   The server context to use.
+     */
+    private static void generateNodeId(@Nonnull ServerContext context) {
+        String currentId = context.getDataStore().get(String.class, "",
```

Comment:
Can we move the setter and getter of this NODE_ID to the server context.
Every component will try to consume it.

Reply:
I think this is not needed now that there is a helper function.